doc: nrf: app_dev: nrf54h20: Global reset (#19608)
nrfconnect/sdk-nrf: doc/nrf/app_dev/device_guides/nrf54h/ug_nrf54h20_debugging.rst

**********************************

To debug single-core applications, you can use the ``west debug`` command to start a single debug session with GDB.

> You can also attach to a running core with ``west attach``.

Reviewer: ```suggestion
You can also attach to a running core using the ``west attach`` command.
```

> You can also attach to a running core using the ``west attach`` command.

Reviewer: ```suggestion
You can also attach the debug session to a running core using the ``west attach`` command.
```

Debugging multi-core applications
*********************************

To debug the firmware running also on cores other than the application core, you need to set up a separate debug session for each one of the cores you want to debug.
When debugging another core, the application core debug session runs in the background and you can debug both cores if needed.

> Debugging early-boot or reset issues

Reviewer: ```suggestion
Debugging early-boot and reset issues
```

> ************************************

Reviewer: ```suggestion
*************************************
```

> A local domain reset triggers a global reset, which in turn resets the debugging state and hence a new debugging connection is needed.
> To debug early boot problems it is possible to trigger a reset and keep the cores in a halted state to give time to attach the debugger, the cores can then be started individually.
> 
> The following sample demonstrates debugging the early boot phase on the application core:

Reviewer: ```suggestion
A local domain reset triggers a global reset, which resets the debugging state. As a result, a new debugging connection is required.
To debug early boot issues, you can trigger a reset while keeping the cores in a halted state. This allows time to attach the debugger. The cores can then be started individually.

The following example demonstrates debugging the early boot phase on the application core:
```

> A local domain reset triggers a global reset, which resets the debugging state. As a result, a new debugging connection is required.
> To debug early boot issues, you can trigger a reset while keeping the cores in a halted state. This allows time to attach the debugger. The cores can then be started individually.
> 
> The following example demonstrates debugging the early boot phase on the application core:

Reviewer: ```suggestion
A local domain reset triggers a global reset, which resets the debugging state. As a result, a new debugging connection is required.
To debug early boot issues, you can trigger a reset while keeping the cores in a halted state. This approach allows time to attach the debugger.
The cores can then be started individually.

The following example demonstrates how to debug the early boot phase on the application core:
```

Reviewer 2: committed
